Source/Forms/Models/Gen5/HiddenGrottoModel.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * This file is part of PokéFinder
+ * Copyright (C) 2017-2021 by Admiral_Fish, bumba, and EzPzStreamz
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "HiddenGrottoModel.hpp"
+#include <Core/Util/Translator.hpp>
+
+HiddenGrottoGeneratorModel5::HiddenGrottoGeneratorModel5(QObject *parent) : TableModel<HiddenGrottoState>(parent)
+{
+}
+
+int HiddenGrottoGeneratorModel5::columnCount(const QModelIndex &parent) const
+{
+    (void)parent;
+    return 5;
+}
+
+QVariant HiddenGrottoGeneratorModel5::data(const QModelIndex &index, int role) const
+{
+    if (role == Qt::DisplayRole)
+    {
+        const auto &state = model[index.row()];
+        int column = index.column();
+        switch (column)
+        {
+        case 0:
+            return QString::number(state.getSeed(), 16).toUpper().rightJustified(16, '0');
```

Comment:
Showing the seed doesn't matter here. Use it to show the chatot pitch.

Reply:
Seed is useful for emu, should I just add another column for the chatot pitch for retail?

Reply:
I prefer consistency. No other generator shows the seed. If the seeds are really necessary this is what researcher is for.

Reply:
Nevermind, noticed you meant for generator specifically. Changed.